core/src/avm2/error.rs

```diff
@@ -34,17 +35,11 @@ impl<'gc> Error<'gc> {
 }
 
 // This type is used very frequently, so make sure it doesn't unexpectedly grow.
-// For now, we only test on Nightly, since a new niche optimization was recently
-// added (https://github.com/rust-lang/rust/pull/94075) that shrinks the size
-// relative to stable.
+#[cfg(target_pointer_width = "32")]
+const _: () = assert!(size_of::<Result<Value<'_>, Error<'_>>>() == 24);
 
-#[rustversion::nightly]
-#[cfg(target_arch = "wasm32")]
```

Comment:
the reason we used wasm32 specifically was because wasm and x86 (specifically ` i686-unknown-linux-gnu`) can have different alignment requirements, and indeed in the past (when we added the assertions originally) the sizes for these two happened to be different because of that. So let's keep `wasm32` here.